packages/core/src/data-centralization/dataCentralizationRuntime.ts

```diff
@@ -96,55 +96,73 @@ export async function dataCentralizationRuntime({
-				{ poolAddress: checksumAddress },
-				{
-					$set: {
-						esXaiRewardRate: updatedPool.averageDailyEsXaiReward,
-						keyRewardRate: updatedPool.averageDailyKeyReward,
-						totalEsXaiClaimed: updatedPool.totalEsXaiClaimed
-					}
-				},
-			);
+					await PoolModel.updateOne(
+						{ poolAddress: checksumAddress },
+						{
+							$set: {
+								esXaiRewardRate: updatedPool.averageDailyEsXaiReward,
+								keyRewardRate: updatedPool.averageDailyKeyReward,
+								totalEsXaiClaimed: updatedPool.totalEsXaiClaimed
+							}
+						},
+					);
+				}
+				const mongoInsertEndTime = new Date().getTime();
+				const totalSeconds = mongoInsertEndTime - startTime;
+				const totalGraphSeconds = graphUpdateEndTime - graphUpdateStartTime;
+				const totalMongoSeconds = mongoInsertEndTime - mongoInsertStartTime;
+				const slackMessage = `Finished pool sync update for ${updatedPools.length} pools in ${totalSeconds}ms. Graph update took ${totalGraphSeconds}ms. Mongo insert took ${totalMongoSeconds}ms.`;
+
+				await sendSlackNotification(slackWebHookUrl, slackMessage, logFunction);
+			} catch (error) {
+				const errorMessage = error instanceof Error ? error.message : String(error);
+				await sendSlackNotification(slackWebHookUrl, `Error in challenge listener: ${errorMessage}`, logFunction);
+				logFunction(`Error in challenge listener: ${errorMessage}`);
+			}
+		},
+		async (error: Error) => {
+			const errorMessage = `Error in listenForChallenges: ${error.message}`;
+			await sendSlackNotification(slackWebHookUrl, errorMessage, logFunction);
+			logFunction(errorMessage);
 		}
-		const mongoInsertEndTime = new Date().getTime();
-		const totalSeconds = mongoInsertEndTime - startTime;
-		const totalGraphSeconds = graphUpdateEndTime - graphUpdateStartTime;
-		const totalMongoSeconds = mongoInsertEndTime - mongoInsertStartTime;
-		const slackMessage = `Finished pool sync update for ${updatedPools.length} pools in ${totalSeconds}ms. Graph update took ${totalGraphSeconds}ms. Mongo insert took ${totalMongoSeconds}ms.`;
-
-		sendSlackNotification(slackWebHookUrl, slackMessage, logFunction);
-	});
+	);
 
 	/**
 	 * Stops the data centralization runtime.
 	 * @returns {Promise<void>} A promise that resolves when the runtime is successfully stopped.
 	 */
 	return async () => {
-		// Disconnect from MongoDB.
-		await mongoose.disconnect();
-		logFunction('Disconnected from MongoDB.');
-		closeChallengeListener();
-		// Remove event listener listener.
-		stopListener();
-		logFunction('Event listener removed.');
+		try {
+			// Disconnect from MongoDB.
+			await mongoose.disconnect();
+			logFunction('Disconnected from MongoDB.');
+			closeChallengeListener();
+			// Remove event listener listener.
+			stopListener();
+			logFunction('Event listener removed.');
+		} catch (error) {
+			logFunction(`Error stopping the data centralization runtime: ${error}`);
+			throw error; // This re-throws the error to ensure it is handled by the caller.
```

Comment:
Throwing in the stop function might lead to unexpected behaviour of the cli app, do we need this here ?